jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/secondLevelCache/InfinispanCacheDeploymentListener.java

```diff
@@ -85,96 +83,88 @@ public Wrapper startCache(Classification classification, Properties properties)
         String container = properties.getProperty(CONTAINER);
         // TODO Figure out how to access CapabilityServiceSupport from here
         ServiceName containerServiceName = ServiceName.parse(InfinispanRequirement.CONTAINER.resolve(container));
+        List<ServiceName> serviceNames = new LinkedList<>();
         EmbeddedCacheManager embeddedCacheManager;
-        ServiceName serviceName;
         if (CACHE_PRIVATE.equals(cache_type)) {
             // need a private cache for non-jpa application use
-            String name = properties.getProperty(NAME);
-            serviceName = ServiceName.JBOSS.append(DEFAULT_CACHE_CONTAINER, (name != null) ? name : UUID.randomUUID().toString());
+            String name = properties.getProperty(NAME, UUID.randomUUID().toString());
+            ServiceName serviceName = ServiceName.JBOSS.append(DEFAULT_CACHE_CONTAINER, name);
+            serviceNames.add(serviceName);
+            ServiceTarget target = currentServiceContainer();
+
+            String[] caches = properties.getProperty(CACHES).split("\\s+");
+            List<ServiceController<Configuration>> controllers = new ArrayList<>(caches.length);
+            for (String cache : caches) {
+                ServiceName aliasServiceName = serviceName.append(cache);
+                ServiceName configServiceName = ServiceName.parse(InfinispanCacheRequirement.CONFIGURATION.resolve(container, cache));
+                controllers.add(new AliasServiceBuilder<>(aliasServiceName, configServiceName, Configuration.class).build(target)
+                        .setInitialMode(ServiceController.Mode.ACTIVE)
+                        .install());
+            }
 
-            ServiceContainer target = currentServiceContainer();
             // Create a mock service that represents this session factory instance
-            ServiceBuilder<EmbeddedCacheManager> builder = new AliasServiceBuilder<>(serviceName, containerServiceName, EmbeddedCacheManager.class).build(target)
+            embeddedCacheManager = new AliasServiceBuilder<>(serviceName, containerServiceName, EmbeddedCacheManager.class).build(target)
                     .setInitialMode(ServiceController.Mode.ACTIVE)
-            ;
-            embeddedCacheManager = ServiceContainerHelper.getValue(builder.install());
+                    .install()
+                    .awaitValue();
 
+            // Ensure cache configuration services are started
+            for (ServiceController<Configuration> controller : controllers) {
+                serviceNames.add(controller.getName());
+                controller.awaitValue();
+            }
         } else {
             // need a shared cache for jpa applications
-            serviceName = containerServiceName;
```

Comment:
Are we purposely not adding the serviceName to serviceNames?

Reply:
Correct.